apps/dapp-console/app/faucet/components/ClaimButton.tsx

```diff
@@ -97,16 +97,21 @@ const ClaimButton = forwardRef<HTMLButtonElement, ClaimButtonProps>(
       if (!authenticated) return
       onClick()
       try {
+        onClaimStarted()
         let response
         if (isOnchainClaim) {
           response = await handleOnchainClaim()
         } else {
           response = await handleOffchainClaim()
         }
 
+        if (response && response.error) {
+          console.error('Claim failed', response.error)
+          onFailed()
```

Comment:
nit: it could be good to return under this line incase we ever have extra logic below this conditional that isn't expecting it to have failed

Reply:
oh yeah, good catch